### PR TITLE
refactor keadm beta join: return error if join failed

### LINKED ISSUE #3874: no error when run `keadm beta join` failed
**What happened**:

no error report when run ` /usr/local/bin/keadm beta join --token= --cloudcore-ipport=x.x.x.x:8080` failed

```console
I0525 16:04:07.755483   32452 command.go:849] 1. Check KubeEdge edgecore process status
I0525 16:04:07.762016   32452 command.go:849] 2. Check if the management directory is clean
F0525 16:04:07.762071   32452 joinbeta.go:91] The management directory /etc/kubeedge/ is not clean, please remove it first
```

**What you expected to happen**:

exit code should not be 0 in this scenario。

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
refactor keadm beta join: return error if join failed
**Which issue(s) this PR fixes**:

Fixes #3874 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/edge/joinbeta.go [hunk 1]
```diff
@@ -64,43 +64,46 @@ func NewJoinBetaCommand() *cobra.Command {
...
 			if running {
-				klog.Exitln("EdgeCore is already running on this node, please run reset to clean up first")
+				return fmt.Errorf("EdgeCore is already running on this node, please run reset to clean up first")
 			}
 
 			step.Printf("Check if the management directory is clean")
 			if _, err := os.Stat(util.KubeEdgePath); err != nil {
 				if os.IsNotExist(err) {
-					return
+					return nil
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/edge/joinbeta.go:80]: should we return err when the directory not exist?

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go:80]: no, this is to ensure the dir /etc/kubeedge does not exist, 
only if the dir exsit, we will report an error

File: keadm/cmd/keadm/app/cmd/edge/joinbeta.go [hunk 2]
```diff
@@ -64,43 +64,46 @@ func NewJoinBetaCommand() *cobra.Command {
...
-				klog.Exitf("The management directory %s is not clean, please remove it first", util.KubeEdgePath)
+				return fmt.Errorf("the management directory %s is not clean, please remove it first", util.KubeEdgePath)
 			}
+			return nil
 		},
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			ver, err := util.GetCurrentVersion(joinOptions.KubeEdgeVersion)
 			if err != nil {
 				klog.Errorf("Edge node join failed: %v", err)
```
Comment (zc2638) [inline keadm/cmd/keadm/app/cmd/edge/joinbeta.go:96]: This may be a redundant log

File: keadm/cmd/keadm/app/cmd/edge/joinbeta.go [hunk 3]
```diff
@@ -64,43 +64,46 @@ func NewJoinBetaCommand() *cobra.Command {
...
-				os.Exit(1)
+				return fmt.Errorf("edge node join failed: %v", err)
 			}
 			joinOptions.KubeEdgeVersion = ver
 
 			if err := join(joinOptions, step); err != nil {
 				klog.Errorf("Edge node join failed: %v", err)
```
Comment (zc2638) [inline keadm/cmd/keadm/app/cmd/edge/joinbeta.go:102]: same

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go]: fixed

**Pull request merged**